lib/config/expression.cpp

```diff
@@ -958,7 +958,17 @@ ExpressionResult ObjectExpression::DoEvaluate(ScriptFrame& frame, DebugHint *dhi
 		name = nameres.GetValue();
 	}
 
-	return VMOps::NewObject(frame, m_Abstract, type, name, m_Filter, m_Zone,
+	if (!m_DefaultTmpl) {
+		if (name.IsEmpty()) {
+			BOOST_THROW_EXCEPTION(ScriptError("Template/object name must not be empty", m_DebugInfo));
```

Comment:
Doesn’t `ObjectExpression#m_DebugInfo` highlight the whole object in the error message, not just its name?

Reply:
Just highlights the `object Host 42` part, not the block following that defines the object properties. So looks fine for me, especially given that it has to handle `object Host {}` as well.